src/main/java/io/vertx/ext/asyncsql/impl/VertxExecutionContext.java

```diff
@@ -0,0 +1,63 @@
+package io.vertx.ext.asyncsql.impl;
+
+import io.vertx.core.Context;
+import io.vertx.core.Handler;
+import io.vertx.core.Vertx;
+import io.vertx.core.logging.Logger;
+import io.vertx.core.logging.LoggerFactory;
+import scala.concurrent.ExecutionContext;
+
+import java.util.Objects;
+
+/**
+ * Execution environment for Scala Future. By default, Scala uses a <em>reasonable default thread pool.</em>. In the
+ * vert.x context, we use the Worker thread pool. If the task is submitted using a Worker thread, the same thread is
+ * used. If not, the task is delegated to a worker thread.
+ *
+ * @author <a href="http://escoffier.me">Clement Escoffier</a>
+ */
+public class VertxExecutionContext implements ExecutionContext {
+
+  public static ExecutionContext create(Vertx vertx) {
+    return new VertxExecutionContext(vertx, (ex) -> LOGGER.error("An exception occurred", ex));
+  }
+
+  public static ExecutionContext create(Vertx vertx,
+                                        Handler<Throwable> handler) {
+    return new VertxExecutionContext(vertx, handler);
+  }
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(VertxExecutionContext.class);
+
+  private final Vertx vertx;
+  private final Handler<Throwable> errorHandler;
+
+  private VertxExecutionContext(Vertx vertx, Handler<Throwable> errorHandler) {
+    Objects.requireNonNull(errorHandler);
+    Objects.requireNonNull(vertx);
+    this.vertx = vertx;
+    this.errorHandler = errorHandler;
+  }
+
+  @Override
+  public void execute(Runnable runnable) {
+    if (Context.isOnWorkerThread()) {
```

Comment:
The async driver is using this execution environment in a lot of place, and as it's _implicit_ it's hard to track down to the real usage. By default, the Scala execution context is a `reasonable default thread pool.`, so I was thinking that it would be safer to be sure to not block the event loop and the code might be blocking.